docs/www/nodule-grid.js

```diff
@@ -377,6 +377,368 @@
+                                //$(sp).parent().parent().addClass("update-data-grandparent");
+                                $("<button class=revert-field title='Revert back to the original data for this field' onclick=revertField(this)><i class='fa fa-arrow-circle-left'></i></button>").insertAfter(sp);
+                            }
+                            if (value == "Lesion with circumscribed margin") {
+                                $(se).prop("checked", true);
+                                $(sp).prop("checked", false);
+                            }
+                            if (value == "Lesion with spiculated margin") {
+                                $(sp).prop("checked", true);
+                                $(se).prop("checked", false);
+                            }
+                        }
+
+                        if (key === "Maximum 2D diameter") {
+                            //console.log("noduleId: theData[noduleId - 1]['Maximum 2D diameter'] or rather value: " + noduleId + " : " + theData[noduleId - 1]["Maximum 2D diameter"] + " : " + value);
+                            currentElement = "#cect" + noduleId + "sl";
+                            if ($(currentElement).hasClass("update-data")) {
+                                $(currentElement).val(value);
+                            } else {
+                                $(currentElement).addClass("update-data");
+                                $(currentElement).attr("original-value", $(currentElement).val());
+                                $(currentElement).val(value);
+                                $("<button class=revert-field title='Revert back to the original data for this field' onclick=revertField(this)><i class='fa fa-arrow-circle-left'></i></button>").insertAfter(currentElement);
+                            }
+                        }
+                        if (key === "Maximum perpendicular 2D diameter") {
+                            //console.log("noduleId: theData[noduleId - 1]['Maximum perpendicular 2D diameter'] or rather value: " + noduleId + " : " + theData[noduleId - 1]["Maximum perpendicular 2D diameter"] + " : " + value);
+                            currentElement = "#cect" + noduleId + "sw";
+                            if ($(currentElement).hasClass("update-data")) {
+                                $(currentElement).val(value);
+                            } else {
+                                $(currentElement).addClass("update-data");
+                                $(currentElement).attr("original-value", $(currentElement).val());
+                                $(currentElement).val(value);
+                                $("<button class=revert-field title='Revert back to the original data for this field' onclick=revertField(this)><i class='fa fa-arrow-circle-left'></i></button>").insertAfter(currentElement);
```

Comment:
These really should not be buttons. I think it's messing with the bootstrap styling when they are added and removed to the DOM. Just use the `<i>` element. Add the onclick and title to that.